src/lib/BeetAPI.js

```diff
@@ -16,7 +17,14 @@ export default class BeetAPI {
                 promise: queued,
                 resolve: qresolve
             });
-            result = await this[request.type](request, vueInst);
+            console.log('waiting');
+            console.log(store.state.WalletStore);
+            if (store.state.WalletStore.isUnlocked==false) {
+                
+                vueInst.$children[2].showAlert(request);
```

Comment:
Bad style `vueInst.$children[2]`

Proper referencing not possible?

Reply:
I know :(

I haven't had a chance to look into it further. Once all popups are fully migrated over to the Popups component which remains constant I can use $refs to get a reference so should be fixed. As it stands, this needs to reference a dynamic router-view so I'm basing it on expected behaviour for now.